src/main/kotlin/com/cognifide/gradle/aem/common/instance/LocalInstance.kt

```diff
@@ -88,7 +88,10 @@ class LocalInstance(aem: AemExtension, name: String) : Instance(aem, name) {
         prop.strings("runModes")?.let { set(it) }
     }
 
-    val runModesString: String get() = (runModes.get() + listOf(purpose.name.lowercase())).joinToString(",")
+    val runModesString: String get() = if (runModes.get().isNotEmpty()) {
```

Comment:
how about single expression and defensive one?

`(runModes.get() + listOf(purpose.name.lowercase())).filterNot { it.isBlank() }.joinToString(",")`

Reply:
yeah that looks better, will change it